src/components/ImageSlider/ImageList.js

```diff
@@ -0,0 +1,243 @@
+import React, { Component } from 'react'
+import { View } from 'react-native'
+import styles from './Styles'
+import Dots from './Dots'
+import Arrow from './Arrow'
+import ImageScrollView from './ImageScrollView'
+import Numbers from './Numbers'
+
+class ImageList extends Component {
+  state = {
+    activeIndex: 0,
+  }
+
+  componentDidMount() {
+    const { enableAutoplay, autoplayTime, editor } = this.props
+    if (!editor && enableAutoplay) {
+      this.autoplay(autoplayTime)
+    }
+  }
+
+  componentWillUnmount() {
+    this.clearAutoplay()
+  }
+
+  clearAutoplay = () => {
+    clearInterval(this.autoplay)
+  }
+
+  autoplay = time => {
+    this.autoplay = setInterval(() => {
+      const { activeIndex } = this.state
+      const { images } = this.props
+      if (activeIndex === images.length - 1) {
+        clearInterval(this.autoplay)
+      } else {
+        this.handleChange(activeIndex + 1)
+      }
+    }, time * 1000)
```

Comment:
1000 is pretty quick. Usually what I see is 2-3 second timers.

Reply:
The app creator can set the timer length, and then it multiplies the time by 1000 to convert the time to milliseconds

Reply:
Oh gotcha. Missed that.

Reply:
Also, FYI autoplay is not working for me. It will auto-transition from the 1st to the 2nd image, but then it will never go to the third. It also looks like it clears the interval when it gets to the end, instead of looping back the beginning (by scrolling back to 0). The seems odd and inconsistent with how these typically work in the real world.

Reply:
The way I have it currently set up, if you press anything the autoplay automatically gets cancelled, and when it gets to the end it also gets cancelled.
I'll change the behavior so it loops back to the beginning if it reaches the end, but can you make sure that you didn't press anything between the 2nd and 3rd image?

Reply:
I definitely didn't press anything.

Going further though, I think we may have misunderstood each other when we were talking about cancelling the interval when the user clicks an arrow, etc. What we should do is when they press an arrow etc. it just resets the timer to the original time. 

## Bad:

![image](https://user-images.githubusercontent.com/955154/90428751-10c9f100-e08a-11ea-9312-1ad2527580ae.png)

## Good:

![image](https://user-images.githubusercontent.com/955154/90428882-453dad00-e08a-11ea-9e7e-c89600f3c83c.png)

Reply:
Yes makes sense, the new changes reflect this.